api/src/main/java/bio/terra/tanagra/underlay/UnderlaysService.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.tanagra.underlay;
+
+import bio.terra.common.exception.NotFoundException;
+import bio.terra.tanagra.app.configuration.UnderlayConfiguration;
+import bio.terra.tanagra.exception.SystemException;
+import bio.terra.tanagra.indexing.FileIO;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** Service to handle underlay operations. */
+@Component
+public class UnderlaysService {
+  private final UnderlayConfiguration underlayConfiguration;
+  private Map<String, Underlay> underlaysMap;
+
+  @Autowired
+  public UnderlaysService(UnderlayConfiguration underlayConfiguration) {
+    this.underlayConfiguration = underlayConfiguration;
+  }
+
+  public List<Underlay> getUnderlays() {
+    return getUnderlaysMap().values().stream().collect(Collectors.toUnmodifiableList());
+  }
+
+  public Underlay getUnderlay(String name) {
+    if (!getUnderlaysMap().containsKey(name)) {
+      throw new NotFoundException("Underlay not found: " + name);
+    }
+    return getUnderlaysMap().get(name);
+  }
+
+  private Map<String, Underlay> getUnderlaysMap() {
+    // TODO: Only read in the underlay config files once at startup, instead of for every call.
```

Comment:
Aren't the underlays already being loaded somewhere else? Can you at least share the code even if you can't avoid loading every time?

Reply:
Ah actually this comment is out of date. We only read them in once. I'll move the reading in to the constructor to make that more explicit.

I can't share code because this is reading in the new underlay config files, and the previous code reads in the yaml files with the proto-generated classes.

Reply:
I was thinking that the indexer reads them too but maybe that code isn't quite the same?

Reply:
Right the call to `Underlay.fromJSON` is made both from here and the indexer, so that's the code reuse. The code you see here mostly handles the differences between the API and the indexer:
- API needs to read in multiple underlays. Indexer only works with one at a time.
- API reads the index files from the JAR `resources`. Indexer typically reads them from disk (though it can also handle reading from the JAR).